internal/pkg/build/files/files_test.go

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) 2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package files
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"reflect"
+	"testing"
+)
+
+/*
+files-test-dir/
+├── dirL1
+│   ├── dirL2
+│   │   ├── dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── .dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── file
+│   │   └── .file
+│   ├── .dirL2
+│   │   ├── dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── .dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── file
+│   │   └── .file
+│   ├── file
+│   └── .file
+├── .dirL1
+│   ├── dirL2
+│   │   ├── dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── .dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── file
+│   │   └── .file
+│   ├── .dirL2
+│   │   ├── dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── .dirL3
+│   │   │   ├── file
+│   │   │   └── .file
+│   │   ├── file
+│   │   └── .file
+│   ├── file
+│   └── .file
+├── file
+└── .file
+
+*/
+
+func createTestDirLayout(t *testing.T) string {
+	testDirName, err := ioutil.TempDir("", "files-test-dir-")
+	if err != nil {
+		t.Fatalf("cannot create temporary dir for testing: %s\n", err)
+	}
+
+	dirList := []string{
```

Comment:
I think this needs entries with:

- blanks in them (spaces, tabs, new lines), both in the filenames and in the intermediate directories
- `*` in them
- `?` in them